### PR TITLE
feat: basic mem reference init

### PR DESCRIPTION
**Pull request by colinnielsen**
# Description

Fill out `VerifierMemReferenceString` initializer

## Link to issue

From #34 

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Refactor (non-breaking change that updates existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Comments have been added/updated

Please delete options that are not relevant.

## Test plan (required)

Demonstrate the code is solid. Which commands did you test with and what are the expected results?
Which tests have you added or updated? Do the tests cover all of the changes included in this PR?

## Screenshots/Screencaps

Please add previews of any UI Changes.

### CODE REVIEW

File: barustenberg/src/srs/reference_string/mem_reference_string.rs [hunk 1]
```diff
@@ -1,17 +1,25 @@
...
 pub(crate) struct VerifierMemReferenceString {
     g2_x: G2Affine,
 }
 
 impl VerifierMemReferenceString {
     pub(crate) fn new(_g2x: &[u8]) -> Self {
-        // Add the necessary code to convert g2x bytes into g2::AffineElement
-        // and initialize precomputed_g2_lines
-        unimplemented!()
+        let g2_x = match G2Affine::deserialize_uncompressed(&*_g2x) {
```
Comment (colinnielsen) [inline barustenberg/src/srs/reference_string/mem_reference_string.rs:13]: @laudiacay do you know if we're using uncompressed affline elements or compressed?

Comment (laudiacay) [reply barustenberg/src/srs/reference_string/mem_reference_string.rs:13]: uhhh grep around in the codebase but i believe i've uniformly used uncompressed! there are a lot of usages in transcript

Comment (colinnielsen) [reply barustenberg/src/srs/reference_string/mem_reference_string.rs:13]: @laudiacay cool. Out of office this week but will sort it Saturday or Monday

Comment (colinnielsen) [reply barustenberg/src/srs/reference_string/mem_reference_string.rs:13]: Yep - this is good to go. added a `from_affline` as well

**Review by 0xJepsen — APPROVED** [review]
LGTM

**Comment by laudiacay** [comment]
i'll merge this once 61 is in (sorry i should have done this earlier) thank you colin

**Comment by laudiacay** [comment]
can you make it so i can push to your fork? or can you rebase off main? then we can merge this, no problem.

**Comment by colinnielsen** [comment]
> can you make it so i can push to your fork? or can you rebase off main? then we can merge this, no problem.

yeah my b

**Comment by colinnielsen** [comment]
> > can you make it so i can push to your fork? or can you rebase off main? then we can merge this, no problem.
> 
> yeah my b

merged and reset

**Review by laudiacay — APPROVED** [review]

**Pull request merged**